index.bs

```diff
@@ -1549,6 +1549,33 @@ See also:
 
 * [[#naming-is-hard]]
 
+<h3 id="overloading">Use overloading wisely</h3>
+
+If the signature or behaviour of a method will be significantly different
+depending on the arguments passed
+you should define a separate method
+rather than overload a single method.
+
+Overloading a method such that one of its arguments can be
+a single thing or an array is useful.
+Similarly, passing a dictionary of options,
+or using optional arguments,
+are good patterns.
+
+However, if subsequent arguments for a method have to change
+because of the first argument passed,
+or if a method has different behaviour depending on the types of inputs,
+this can hinder code readability,
+and make discovering API functionality more difficult for authors,
+and should be avoided.
+
+<div class="example">
+TODO
+https://xhr.spec.whatwg.org/#interface-formdata
+https://html.spec.whatwg.org/multipage/canvas.html#canvasdrawpath
+https://html.spec.whatwg.org/multipage/canvas.html#canvasdrawimage
```

Comment:
Yikes, this one is really bad. We could also use it as an example of what not to do for https://w3ctag.github.io/design-principles/#prefer-dict-to-bool

Reply:
Yeah, maybe this is a better example for that principle than this one?